ckan/config/middleware/flask_app.py

```diff
@@ -327,7 +327,10 @@ def load_user_from_request(request):  # type: ignore
         This callback function is called whenever a user could not be
         authenticated via the session cookie, so we fall back to the API token.
         """
+        g.login_via_request = True
```

Comment:
A better name might reduce the need for comments below
```suggestion
        g.login_via_auth_header = True
```